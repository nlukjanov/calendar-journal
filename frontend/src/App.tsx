import React, { lazy, Suspense } from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';

const Landing = lazy(() => import('./components/Landing/Landing'));
const Navbar = lazy(() => import('./components/Navbar/Navbar'));
const Signup = lazy(() => import('./components/Signup/Signup'));
const Signin = lazy(() => import('./components/Signin/Signin'));
const MyJournal = lazy(() => import('./components/MyJournal/MyJournal'));
const NewEntry = lazy(() => import('./components/NewEntry/NewEntry'));
const ErrorPage = lazy(() => import('./components/ErrorPage/ErrorPage'));

const SecureRouteFront = lazy(() => import('./lib/SecureRouteFront'));

function App() {
  return (
    <Suspense fallback={<div>Loading...</div>}>
      <BrowserRouter>
        <main data-testid='main'>
          <Navbar />
          <Switch>
            <Route exact path='/' component={Landing} />
            <Route path='/signup' component={Signup} />
            <Route path='/signin' component={Signin} />
            <SecureRouteFront path='/myjournal' component={MyJournal} />
            <SecureRouteFront
              path='/new-entry'
              component={(props: any) => {
                return <NewEntry {...props} />;
              }}
            />
            <Route path='/*' component={ErrorPage} />
          </Switch>
        </main>
      </BrowserRouter>
    </Suspense>
  );
}

export default App;
