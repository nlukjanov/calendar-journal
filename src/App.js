import React, { lazy, Suspense } from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import './App.css';

const Landing = lazy(() => import('./components/Landing/Landing'));
const Navbar = lazy(() => import('./components/Navbar/Navbar'));
const Signup = lazy(() => import('./components/Signup/Signup'));
const Signin = lazy(() => import('./components/Signin/Signin'));
const MyJournal = lazy(() => import('./components/MyJournal/MyJournal'));
const NewEntry = lazy(() => import('./components/NewEntry/NewEntry'));


function App() {
  return (
    <Suspense fallback={<div>Loading...</div> }>
      <BrowserRouter>
        <main data-testid='main'>
          <Navbar />
          <Switch>
            <Route exact path='/' component={Landing} />
            <Route path='/signup' component={Signup} />
            <Route path='/signin' component={Signin} />
            <Route path='/myjournal' component={MyJournal} />
            <Route path='/new-entry' component={NewEntry} />
          </Switch>
        </main>
      </BrowserRouter>
    </Suspense>
  );
}

export default App;
