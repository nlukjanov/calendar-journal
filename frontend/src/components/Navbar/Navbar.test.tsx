import React from 'react';

import { render, waitFor } from '@testing-library/react';
import App from '../../App';

test('renders all buttons on navbar', async () => {
  const { getByText, getByTestId, queryByText, container } = render(<App />);
  await waitFor(() => expect(getByText(/Home/i)).toBeInTheDocument());
  expect(getByTestId('nav-signin')).toHaveTextContent('Sign In');
  expect(getByTestId('nav-signup')).toHaveTextContent('Sign Up');
  expect(queryByText(/New Entry/i)).not.toBeInTheDocument();
  expect(queryByText(/Logout/i)).not.toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
