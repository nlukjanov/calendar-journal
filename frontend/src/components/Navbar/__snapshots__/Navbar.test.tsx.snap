// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all buttons on navbar 1`] = `
<div>
  <main
    data-testid="main"
    style=""
  >
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <a
        href="/"
      >
        <span
          class="navbar-brand"
        >
          Home
        </span>
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="justify-content-end navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="navbar-nav"
        >
          <a
            class="nav-link"
            data-testid="nav-signin"
            href="/signin"
          >
            Sign In
          </a>
          <a
            class="nav-link"
            data-testid="nav-signup"
            href="/signup"
          >
            Sign Up
          </a>
        </div>
      </div>
    </nav>
    <div
      class="jumbotron"
    >
      <div
        class="text-center container"
      >
        <h1>
          Welcome to Calendar Journal
        </h1>
        <p>
          Take your notes and keep your memories
        </p>
        <div
          class="d-flex justify-content-center"
        >
          <a
            class="nav-link"
            href="/signin"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Sign In
            </button>
          </a>
          <a
            class="nav-link"
            href="/signup"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Sign Up
            </button>
          </a>
        </div>
      </div>
    </div>
  </main>
</div>
`;
